.github/workflows/cypress-test-multidatasources-enabled-e2e.yml

```diff
@@ -0,0 +1,120 @@
+name: E2E multi datasources enabled workflow
+
+on: [ push, pull_request ]
+
+env:
+  OPENSEARCH_VERSION: '3.0.0'
+  CI: 1
+  # avoid warnings like "tput: No value for $TERM and no -T specified"
+  TERM: xterm
+  PLUGIN_NAME: opensearch-security
+  OPENSEARCH_INITIAL_ADMIN_PASSWORD: myStrongPassword123!
+  
+jobs:
+  tests:
+    name: Run Cypress multidatasources tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout Branch
+        uses: actions/checkout@v3
+
+      - name: Set env
+        run: |
+          opensearch_version=$(node -p "require('./package.json').opensearchDashboards.version")
+          plugin_version=$(node -p "require('./package.json').version")
+          echo "OPENSEARCH_VERSION=$opensearch_version" >> $GITHUB_ENV
+          echo "PLUGIN_VERSION=$plugin_version" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Create remote OpenSearch Config
+        if: ${{ runner.os == 'Linux' }}
+        run: | 
+          cat << 'EOT' > remote_opensearch.yml
+          http.port: 9202
+          plugins.security.ssl.transport.pemcert_filepath: esnode.pem
+          plugins.security.ssl.transport.pemkey_filepath: esnode-key.pem
+          plugins.security.ssl.transport.pemtrustedcas_filepath: root-ca.pem
+          plugins.security.ssl.transport.enforce_hostname_verification: false
+          plugins.security.ssl.http.pemcert_filepath: esnode.pem
+          plugins.security.ssl.http.pemkey_filepath: esnode-key.pem
+          plugins.security.ssl.http.pemtrustedcas_filepath: root-ca.pem
+          plugins.security.allow_unsafe_democertificates: true
+          plugins.security.allow_default_init_securityindex: true
+          plugins.security.authcz.admin_dn:
+            - 'CN=A,OU=UNIT,O=ORG,L=TORONTO,ST=ONTARIO,C=CA'
+          plugins.security.nodes_dn:
+            - 'CN=node1.dns.a-record,OU=UNIT,O=ORG,L=TORONTO,ST=ONTARIO,C=CA'
+            - 'CN=node2.dns.a-record,OU=UNIT,O=ORG,L=TORONTO,ST=ONTARIO,C=CA'
+          plugins.security.audit.type: internal_opensearch
+          plugins.security.enable_snapshot_restore_privilege: true
+          plugins.security.check_snapshot_restore_write_privileges: true
+          # TODO: change this back to true/just append to the created opensearch.yml the new port
+          # after the self-signed certs issue is fixed
+          plugins.security.ssl.http.enabled: false 
+          plugins.security.restapi.roles_enabled: [all_access, security_rest_api_access]
+          plugins.security.system_indices.enabled: true
+          plugins.security.system_indices.indices: [.plugins-ml-config, .plugins-ml-connector,
```

Comment:
Is there any way we can start with the demo security config as a baseline and then make modifications to it like setting the port as 9202 and disabling https (for now)?

This list is subject to change and ideally should be defined in a single place.

Reply:
Yes - I did so in: https://github.com/derek-ho/start-opensearch/blob/v3/action.yml#L132 for OpenSearch's running without security, just by adding a port number that is not standard. However the issue is that the security demo configuration by default writes https enabled, and you cannot simply add on a disable block of the same setting, since it would complain about duplicate fields. Thus as a workaround, this is a different branch where I instead replace the whole opensearch.yml. Once your fix goes into core, we can re-enable https and simply add a port number to the opensearch.yml after the security demo configuration, instead of rewriting the whole thing. Going to mark this as resolved for now, but feel free to re-open if that didn't cover your concern.